.github/workflows/deploy-dagster-cloud.yml

```diff
@@ -130,83 +185,75 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           context: ./hooli-data-ingest
           push: true
           tags: ${{ env.IMAGE_REGISTRY }}:${{ env.IMAGE_TAG }}-hooli-data-ingest
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
 
       - name: Update build session with image tag for hooli_data_ingest
         id: ci-set-build-output-hooli-data-ingest
-        if: steps.prerun.outputs.result != 'skip'
-        uses: dagster-io/dagster-cloud-action/actions/utils/dagster-cloud-cli@v0.1.38
+        if: steps.prerun.outputs.result != 'skip'  && steps.changed-files.outputs.hooli-data-ingest_any_changed == 'true'  
+        uses: dagster-io/dagster-cloud-action/actions/utils/dagster-cloud-cli@v0.1.47
         with:
           command: "ci set-build-output --location-name=hooli_data_ingest --image-tag=$IMAGE_TAG-hooli-data-ingest"
 
       # Build 'hooli_bi' code location
       - name: Build and upload Docker image for hooli_bi
-        if: steps.prerun.outputs.result != 'skip'
+        if: steps.prerun.outputs.result != 'skip' && steps.changed-files.outputs.hooli-bi_any_changed == 'true'  
         uses: docker/build-push-action@v5
         with:
           context: ./hooli-bi
           push: true
           tags: ${{ env.IMAGE_REGISTRY }}:${{ env.IMAGE_TAG }}-hooli-bi
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
 
       - name: Update build session with image tag for hooli_bi
         id: ci-set-build-output-hooli-bi
-        if: steps.prerun.outputs.result != 'skip'
-        uses: dagster-io/dagster-cloud-action/actions/utils/dagster-cloud-cli@v0.1.38
+        if: steps.prerun.outputs.result != 'skip' && steps.changed-files.outputs.hooli-bi_any_changed == 'true'  
+        uses: dagster-io/dagster-cloud-action/actions/utils/dagster-cloud-cli@v0.1.47
         with:
           command: "ci set-build-output --location-name=hooli_bi --image-tag=$IMAGE_TAG-hooli-bi"
 
       # Build pipes example container
       - name: Build and upload Docker image for pipes example
-        if: steps.prerun.outputs.result != 'skip'
+        if: steps.prerun.outputs.result != 'skip' && steps.changed-files.outputs.hooli_data_eng_any_changed == 'true'  
         uses: docker/build-push-action@v5
         with:
           context: ./hooli_data_eng/utils/example_container
           push: true
           tags: ${{ env.IMAGE_REGISTRY }}:latest-pipes-example
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
 
-      # Deploy
+      #Deploy
       - name: Deploy to Dagster Cloud
         id: ci-deploy
         if: steps.prerun.outputs.result != 'skip'
-        uses: dagster-io/dagster-cloud-action/actions/utils/dagster-cloud-cli@v0.1.38
+        uses: dagster-io/dagster-cloud-action/actions/utils/dagster-cloud-cli@v0.1.47
         with:
-          command: "ci deploy"
+          command: "ci deploy $LOCATIONS"
```

Comment:
If we are manually constructing the locations flag here, I wonder if we could remove all of the condition `ci set build output` calls? We may need to also manually add the image tag or something... I'm not 100% sure what the build output contains, but it feels like we're setting up for a bit of confusion here (kind of two sources of what to deploy)

Reply:
oh that's a really interesting idea -- I like the idea, will play around with it

Reply:
ok so I tried doing `ci build` vs. `ci deploy` -- build expects to build the dockerfiles though, which we do in the previous steps.

I ended up consolidating the build-set-output steps to the end so you build-set-output based on the same underlying inputs as the ci deploy step

Reply:
so it turns out that you can do `dagster-cloud ci deploy --location-name data-eng-pipeline --location-name basics` works, but `dagster-cloud ci set-build-output --location-name data-eng-pipeline --image-tag {image_tag_a}  --location-name basics --image-tag {image_tag_b}` does not (it just uses the last image tag value for all of the locations 🤔 ).

Anyway, I pivoted to called the `dagster-cloud` CLI directly, still using the same inputs as the `dagster-cloud deploy`, so I think it's an improvement. Since I needed to install the `dagster-cloud` package I also added a uv.lock for that and am using the uv cache so that it shouldn't need to be re-installed unless that uv lock changes.